### ISSUE TITLE
Erratic results when running legacy programs that do single character read

### ISSUE THREAD

**Issue by denhamson** [intersystems-ru/webterminal#87]
Running a legacy program that executes command   Read *IN:TIMEOUT appears to work OK for alphabetic and numeric characters but returns strange results with punctuation characters.  Examples "/" character input echoes as "¿" and "*" character input echoes as "8" with corresponding wrong ascii values in the IN variable.

**Comment by eduard93** [comment]
Unable to reproduce. 

```
USER > r a:2
/
USER > w a 
 /
```

Please provide code to reproduce.

**Comment by denhamson** [comment]
See attached export of a object script routine called readtest and attached sample output.

For all three read types the keys entered were / then * then ENTER  

Hope this helps. 
 
Mike Brown
mike.brown.mbcs@me.com

**Comment by denhamson** [comment]
Apologies Forgot the attachments


Mike Brown
mike.brown.mbcs@me.com

**Comment by ZitRos** [comment]
@denhamson, attachments are still not attached (I guess the mail response doesn't work for attachments here). Thanks!

**Comment by denhamson** [comment]
Trying with pdf attachments, GitHub doesn’t appear to look xml or rtf file types.

Mike Brown
mike.brown.mbcs@me.com

**Comment by denhamson** [comment]
You are correct, GitHub doesn’t pick up attachments from emails, I have uploaded attachments directly to GitHub website.
Mike Brown
mike.brown.mbcs@me.com

**Comment by ZitRos** [comment]
@denhamson, I still cannot see the attachments. Try using zip for other file types. You need just to drag&drop file onto the comment input here on GitHub.

**Comment by denhamson** [comment]
Done

Mike Brown
mike.brown.mbcs@me.com

**Comment by ZitRos** [comment]
@denhamson, there is still no attachments. Please go directly to GitHub here: https://github.com/intersystems-ru/webterminal/issues/87 and upload the files (zip).

**Comment by denhamson** [comment]
attachments

[readtest.xml.pdf](https://github.com/intersystems-ru/webterminal/files/736858/readtest.xml.pdf)
[readtestSampleOutput.pdf](https://github.com/intersystems-ru/webterminal/files/736859/readtestSampleOutput.pdf)

**Comment by eduard93** [comment]
```
testreadsingle
	set timeout=4 ; four second timeout
	read !,"single>",*in:timeout
	goto:in=13 testreadfixed ; exit when empty response
	write !,"in=",in,?10,"$zb=",$zb,?20,"$za=",$za,?30,"$test=",$test
	goto testreadsingle
testreadfixed
	read !,"fixed>",in#1:timeout
	goto:in="" testreadnormal ; exit when empty response
	write !,"in=",in,?10,"$zb=",$zb,?20,"$za=",$za,?30,"$test=",$test
	goto testreadfixed
testreadnormal
	read !,"normal>",in:timeout
	quit:in="" ; quit when empty response
	write !,"in=",in,?10,"$zb=",$zb,?20,"$za=",$za,?30,"$test=",$test
	goto testreadnormal

```
